libs/ui/src/lib/notification/Modal/ModalProps.ts

```diff
@@ -0,0 +1,51 @@
+export type ModalTypesProps = 'dialog' | 'alert' | 'blank';
+
+export type ModalProps = {
+  id?: string;
+  type: ModalTypesProps;
+  title: string;
+  children: React.ReactNode;
+  disableBackdropClick?: boolean;
+  opened?: boolean;
+  confirm?: {
+    label: string;
+    handler?: () => void;
+  };
+  dismiss?: {
+    label: string;
+    handler?: () => void;
+  };
+};
+
+export type BlankModalProps = Omit<
+  ModalProps,
+  'confirm' | 'dismiss' | 'type' | 'children' | 'opened'
```

Comment:
Como o `'confirm' | 'dismiss' | 'opened'` são opcionais na tipagem `ModalProps`, não sei se você precisa realmente omiti-los aqui.

```suggestion
  'type' | 'children'
```

Reply:
O mesmo vale para as demais funções que estão fazendo a mesma coisa com outros atributos.

Reply:
Tanto o tipo `BlankModalProps`, quanto os tipos `AlertModalProps` e `DialogModalProps` eu uso para as funções `addModal`, `addAlert` e `addDialog` do hook `useModal`. Mesmo que o modal no contexto geral aceite essas props como opcional, essas funções não devem aceitar por não fazer uso deles.

Reply:
Quanto ao `Pick` ele faz sentido para o `BlankModalProps` mas eu mantive com o `Omit` para deixar na mesma "linguagem" dos outros dois e evitar confusão.